### PR TITLE
Initialize async plasma for async actors

### LINKED ISSUE #6801: Async init shouldn't be required inside async actor
```py3
---------------------------------------------------------------------------
RayTaskError(AssertionError)              Traceback (most recent call last)
<ipython-input-17-3d607351d5d7> in <module>
----> 1 ray.get(a.something.remote())

~/Desktop/ray/ray/python/ray/worker.py in get(object_ids, timeout)
   1480                     worker.core_worker.dump_object_store_memory_usage()
   1481                 if isinstance(value, RayTaskError):
-> 1482                     raise value.as_instanceof_cause()
   1483                 else:
   1484                     raise value

RayTaskError(AssertionError): ray::AsyncActor.something() (pid=26320, ip=10.142.32.76)
  File "python/ray/_raylet.pyx", line 643, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 621, in function_executor
  File "/Users/simonmo/miniconda3/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/Users/simonmo/miniconda3/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "<ipython-input-6-b3532e96dec8>", line 7, in something
  File "python/ray/includes/unique_ids.pxi", line 199, in ray._raylet.ObjectID.__await__
  File "/Users/simonmo/Desktop/ray/ray/python/ray/async_compat.py", line 38, in get_async
    async_api_init()
  File "/Users/simonmo/Desktop/ray/ray/python/ray/experimental/async_api.py", line 98, in init
    "You are using async_api inside a running event loop. "
AssertionError: You are using async_api inside a running event loop. Please call `await _async_init()` to initialize inside asynchrounous context.


In [20]: @ray.remote
    ...: class AsyncActor:
    ...:     async def something(self):
    ...:         import asyncio
    ...:         await _async_init()
    ...:         await asyncio.sleep(0.)
    ...:         print("Done")
    ...:         result = await task.remote() <- somethow this requires asyncio API to be initialized
    ...:         print(result)
    ...:         ray.get(task.remote())
```

### PR DESCRIPTION
**Pull request by simon-mo**
fix #6801

### CODE REVIEW

File: python/ray/tests/py3_test.py [hunk 1]
```diff
@@ -247,3 +247,17 @@ def throw_error(self):
...
+    @ray.remote
+    def remote_task():
+        return 1
+
+    @ray.remote
+    class AsyncGetter:
+        async def get(self):
+            return await remote_task.remote()
+
+    getter = AsyncGetter.options(is_direct_call=True, is_asyncio=True).remote()
```
Comment (edoakes) [inline python/ray/tests/py3_test.py]: Don't need `is_direct_call=True` anymore, it's on by default.

File: python/ray/_raylet.pyx [hunk 2]
```diff
@@ -1173,6 +1173,10 @@ cdef class CoreWorker:
         if self.async_event_loop is None:
             self.async_event_loop = asyncio.new_event_loop()
             asyncio.set_event_loop(self.async_event_loop)
+            # Initialize the async plasma connection in worker
+            # Delayed import due to everything depends on raylet
```
Comment (edoakes) [inline python/ray/_raylet.pyx]: I don't understand this comment out of context. What is "everything" and does it depend on the raylet process or some other raylet-related import?

File: python/ray/_raylet.pyx [hunk 3]
```diff
@@ -1173,6 +1173,10 @@ cdef class CoreWorker:
         if self.async_event_loop is None:
             self.async_event_loop = asyncio.new_event_loop()
             asyncio.set_event_loop(self.async_event_loop)
+            # Initialize the async plasma connection in worker
```
Comment (edoakes) [inline python/ray/_raylet.pyx]:
```suggestion
            # Initialize the async plasma connection.
```

File: python/ray/_raylet.pyx (hunk 2 above)
Comment (edoakes) [inline python/ray/_raylet.pyx]:
```suggestion
            # Delayed import due to everything depends on raylet.
```

File: python/ray/tests/py3_test.py [hunk 4]
```diff
@@ -247,3 +247,17 @@ def throw_error(self):
 
     with pytest.raises(ray.exceptions.RayTaskError):
         await ray.async_compat.get_async(direct.throw_error.remote())
+
+
+def test_asyncio_actor_async_get(ray_start_regular_shared):
```
Comment (edoakes) [inline python/ray/tests/py3_test.py]: Is this the only test for asyncio actors right now? If so, could you add some more tests?

**Comment by simon-mo** [comment]
> Is this the only test for asyncio actors right now? If so, could you add some more tests? 

There are more tests in the same file:
```
  <Function test_asyncio_actor>
  <Function test_asyncio_actor_same_thread>
  <Function test_asyncio_actor_concurrency>
  <Function test_asyncio_actor_high_concurrency>
  <Function test_asyncio_get>
  <Function test_asyncio_actor_async_get>
```

**Comment by simon-mo** [comment]
> Is this the only test for asyncio actors right now? If so, could you add some more tests?

**Review by edoakes — APPROVED** [review]
LGTM

**Pull request merged**